core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -176,8 +186,61 @@ abstract class AppBase(
 
   var driverAccumMap: HashMap[Long, ArrayBuffer[DriverAccumCase]] =
     HashMap[Long, ArrayBuffer[DriverAccumCase]]()
-  var mlEventLogType = ""
-  var pysparkLogFlag = false
+
+
+  // Returns the String value of the eventlog or empty if it is not defined. Note that the eventlog
+  // won't be defined for running applications
+  def getEventLogPath: String = {
+    eventLogInfo.map(_.eventLog).getOrElse(new Path("")).toString
+  }
+
+  // Update the endTime of the application and calculate the duration.
+  // This is called while handling ApplicationEnd event
+  def updateEndTime(newEndTime: Long): Unit = {
+    appMetaData.foreach(_.setEndTime(newEndTime))
+  }
+
+  // Returns a boolean flag to indicate whether the endTime was estimated.
+  def isAppDurationEstimated: Boolean = {
```

Comment:
Nit: This can be simplified similiar to the below function getAppName. I am fine in the current form as well.